CMakeLists.txt

```diff
@@ -27,14 +27,24 @@ add_definitions(-DAPI_NAME="Vulkan")
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 find_package(PythonInterp 3 REQUIRED)
 
-find_package(VulkanHeaders)
-if(NOT ${VulkanHeaders_FOUND})
-    message(FATAL_ERROR "Could not find Vulkan headers path. This can be fixed by setting VULKAN_HEADERS_INSTALL_DIR to an "
-                        "installation of the Vulkan-Headers repository.")
-endif()
-if(NOT ${VulkanRegistry_FOUND})
-    message(FATAL_ERROR "Could not find Vulkan registry path. This can be fixed by setting VULKAN_HEADERS_INSTALL_DIR to an "
-                        "installation of the Vulkan-Headers repository.")
+if (TARGET Vulkan::Headers)
+    message(STATUS "Using Vulkan headers from Vulkan::Headers target")
+    get_target_property(VulkanRegistry_DIR Vulkan::Registry INTERFACE_INCLUDE_DIRECTORIES)
+else()
+    find_package(VulkanHeaders)
+    if(NOT ${VulkanHeaders_FOUND})
+        message(FATAL_ERROR "Could not find Vulkan headers path. This can be fixed by setting VULKAN_HEADERS_INSTALL_DIR to an "
+                            "installation of the Vulkan-Headers repository.")
+    endif()
+    if(NOT ${VulkanRegistry_FOUND})
+        message(FATAL_ERROR "Could not find Vulkan registry path. This can be fixed by setting VULKAN_HEADERS_INSTALL_DIR to an "
+                            "installation of the Vulkan-Headers repository.")
+    endif()
+
+    # set up the Vulkan::Headers target for consistency
```

Comment:
This block should be moved up, nearer the Header setup.

Reply:
I'm not sure I follow. This block needs to run after find_package(VulkanHeaders), so I don't think we can move it up. Can you clarify?